vasiliy/src/main/java/data_structures/MyLinkedList.java

```diff
@@ -1,6 +1,207 @@
+        if (!checkIndex(index)) {
+            return false;
+        }
+        Node objectNode = new Node(o);
+        Node node;
+        if (index == size - 1) {
+            node = tail;
+        } else {
+            node = findNode(index);
+        }
+        node.setPrevious(objectNode);
+        objectNode.setNext(node);
+        if (index != 0) {
+            objectNode.setPrevious(node.getPrevious());
+            node.getPrevious().setNext(objectNode);
+        } else {
+            head = objectNode;
+        }
+        size++;
+        return true;
+    }
+
+    private Node findNode(int index) {
+        if (!checkIndex(index) || isEmpty()) {
+            return null;
+        }
+        if (index == size - 1) {
+            return tail;
+        }
+        Node node = head;
+        for (int i = 0; i < index; i++) {
+            node = node.getNext();
+        }
+        return node;
+    }
+
+    @Override
+    public Object get(int index) {
+        if (!checkIndex(index)) {
+            return null;
+        }
+        return findNode(index).getValue();
+    }
+
+    private void removeNode(Node node) {
+        if (size == 1) {
+            clear();
+            return;
+        }
+        if (node == head) {
+            node.getNext().setPrevious(null);
+            head = node.getNext();
+            node.setNext(null);
+            return;
+        }
+        if (node == head) {
+            node.getPrevious().setNext(null);
+            tail = node.getPrevious();
+            node.setPrevious(null);
+            return;
+        }
+        node.getPrevious().setNext(node.getNext());
+        node.getNext().setPrevious(node.getPrevious());
+        node.setNext(null);
+        node.setPrevious(null);
+        size--;
+    }
+
+    @Override
+    public Object remove(int index) {
+        if (!checkIndex(index)) {
+            return null;
+        }
+        Node node = findNode(index);
+        removeNode(node);
+        return node.getValue();
+    }
+
+    private Node findNode(Object o) {
+        if (isEmpty()) {
+            return null;
+        }
+        if (o == null) {
+            Node node = head;
+            do {
+                if (node.getValue() == null) {
+                    return node;
+                }
+                node = node.getNext();
+            } while (node != null);
+        } else {
+            Node node = head;
+            do {
+                if (o.equals(node.getValue())) {
+                    return node;
+                }
+                node = node.getNext();
+            } while (node != null);
+        }
+        return null;
+    }
+
+    @Override
+    public boolean remove(Object o) {
+        Node node = findNode(o);
+        if (node == null) {
+            return false;
+        }
+        removeNode(node);
+        return true;
+    }
+
+    @Override
+    public Object set(int index, Object o) {
+        if (!checkIndex(index)) {
```

Comment:
You will duplicate this check in method:
private Node findNode(int index) {
        if (!checkIndex(index) || isEmpty()) {
            return null;
        }